client/components/catalog/Card/Tags/index.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+  <div>
+    <v-chip
+      v-for="tag in repository.tags"
+      :key="tag.id"
+      @click:close="showDeleteConfirmation(tag)"
+      color="grey darken-1"
+      text-color="white"
+      close>
+      {{ tagName(tag) }}
+    </v-chip>
+    <add-tag v-if="exceedTagLimit" :repository="repository" />
+  </div>
+</template>
+
+<script>
+import AddTag from './AddTag';
+import EventBus from 'EventBus';
+import { mapActions } from 'vuex';
+import truncate from 'lodash/truncate';
+
+const appChannel = EventBus.channel('app');
+const TAG_LIMIT = 3;
+
+export default {
+  props: {
+    repository: { type: Object, required: true }
+  },
+  computed: {
+    exceedTagLimit: ({ repository }) => repository.tags.length < TAG_LIMIT
+  },
+  methods: {
+    ...mapActions('repositories', ['removeTag']),
+    showDeleteConfirmation(tag) {
+      const data = { repositoryId: this.repository.id, tagId: tag.id };
+      appChannel.emit('showConfirmationModal', {
+        title: 'Delete tag',
+        message: `Are you sure you want to delete tag ${tag.name}?`,
+        action: () => this.removeTag(data)
+      });
+    },
+    tagName(tag) {
```

Comment:
You don't need to check tag.name length (truncate will do that for you; it won't truncate if it is bigger). Method name should start with verb i.e. truncateTagName. Probably if there is only one tag it will not fit without truncation (so we also need default truncation length). You can simplify this entiely with something like
```
const length = [15, 6, 5][repository.tags.length - 1];
return truncate(tag.name, { length }));
```